python_modules/dagster/dagster/_core/definitions/declarative_scheduling/slice_condition.py

```diff
@@ -0,0 +1,52 @@
+from abc import abstractmethod
+
+from dagster._core.asset_graph_view.asset_graph_view import AssetSlice
+from dagster._core.definitions.asset_subset import ValidAssetSubset
+
+from .asset_condition import AssetCondition, AssetConditionResult
+from .scheduling_condition_evaluation_context import SchedulingConditionEvaluationContext
+
+
+class SliceSchedulingCondition(AssetCondition):
+    """Base class for simple conditions which compute a simple slice of the asset graph."""
+
+    @abstractmethod
+    def compute_slice(self, context: SchedulingConditionEvaluationContext) -> AssetSlice: ...
+
+    def evaluate(self, context: SchedulingConditionEvaluationContext) -> AssetConditionResult:
+        # don't compute anything if there are no candidates
+        if context.candidate_slice.is_empty:
+            true_slice = context.asset_graph_view.create_empty_slice(context.asset_key)
+        else:
+            true_slice = self.compute_slice(context)
+
+        return AssetConditionResult.create(context, true_slice.convert_to_valid_asset_subset())
+
+
+class MissingSchedulingCondition(SliceSchedulingCondition):
+    @property
+    def description(self) -> str:
+        return "Missing"
+
+    def compute_slice(self, context: SchedulingConditionEvaluationContext) -> AssetSlice:
+        # TODO: this logic should be simplified once we have a unified way of detecting both
+        # materializations and observations through the parittion status cache
+        if context.asset_graph_view.asset_graph.get(context.asset_key).is_materializable:
+            # cheap call which takes advantage of the partition status cache
+            materialized_slice = context.asset_graph_view.compute_materialized_asset_slice(
```

Comment:
Oh i see you call it here.

I still think having a function called this in the asset graph view layer is confusing.

I think we should push the logic in `MissingSchedulingCondition` into the AGV.